services/ble/GattClient.hpp

```diff
@@ -45,22 +45,23 @@ namespace services
         using infra::Observer<GattClientCharacteristicOperationsObserver, GattClientCharacteristicOperations>::Observer;
 
         virtual AttAttribute::Handle CharacteristicValueHandle() const = 0;
-        virtual GattCharacteristic::PropertyFlags CharacteristicProperties() const = 0;
     };
 
     class GattClientCharacteristicOperations
         : public infra::Subject<GattClientCharacteristicOperationsObserver>
         , public infra::Subject<GattClientStackUpdateObserver>
     {
     public:
-        virtual void Read(const GattClientCharacteristicOperationsObserver& characteristic, const infra::Function<void(const infra::ConstByteRange&)>& onRead) = 0;
-        virtual void Write(const GattClientCharacteristicOperationsObserver& characteristic, infra::ConstByteRange data, const infra::Function<void()>& onDone) = 0;
+        using OperationOnDone = infra::Function<void(uint8_t)>;
```

Comment:
What is the semantics of the uint8_t parameter? This seems to be the right place to document it. By the way, the syntax allows for a named argument: infra::Function<void(uint8_t cubicFeet)>;